diag_manager/fms_diag_object.F90

```diff
@@ -568,60 +591,19 @@ logical function fms_diag_accept_data (this, diag_field_id, field_data, time, is
     call this%FMS_diag_fields(diag_field_id)%set_data_buffer_is_allocated(.TRUE.)
     call this%FMS_diag_fields(diag_field_id)%set_math_needs_to_be_done(.TRUE.)
 !$omp end critical
+    !TODO Save the field_weight and the oor_mask to use later in the calculations
     call this%FMS_diag_fields(diag_field_id)%set_data_buffer(field_data,&
                                                              is, js, ks, ie, je, ke)
     fms_diag_accept_data = .TRUE.
     return
   else
-!!TODO: Loop through fields and do averages/math functions
-
     call this%allocate_diag_field_output_buffers(field_data, diag_field_id)
-    do i = 1, size(this%FMS_diag_fields(diag_field_id)%buffer_ids)
-      buffer_id = this%FMS_diag_fields(diag_field_id)%buffer_ids(i)
-
-      !!TODO: Check if the field is a physics window
-      !! phys_window = fms_diag_compare_window()
-
-      !!TODO: Get local start and end indices on 3 axes for regional output
-
-      !> Compute the diurnal index
-      sample = 1
-      if (present(time)) then
-        call get_time(time, second, day, tick) !< Current time in days and seconds
-        ptr_diag_field_yaml => diag_yaml%get_diag_field_from_id(buffer_id)
-        sample = floor((second + real(tick) / get_ticks_per_second()) &
-          & * ptr_diag_field_yaml%get_n_diurnal() / SECONDS_PER_DAY) + 1
-      end if
-
-      !!TODO: Get the vertical layer start and end indices
-
-      !!TODO: Initialize output time for fields output every time step
-
-      !< Check if time should be present for this field
-      if (.not.this%FMS_diag_fields(diag_field_id)%is_static() .and. .not.present(time)) then
-        write(error_string, '(a,"/",a)') trim(this%FMS_diag_fields(diag_field_id)%get_modname()),&
-          & trim(this%FMS_diag_fields(diag_field_id)%diag_field(i)%get_var_outname())
-        if (fms_error_handler('fms_diag_object_mod::fms_diag_accept_data', 'module/output_name: '&
-          &//trim(error_string)//', time must be present for nonstatic field', err_msg)) then
-            !!TODO: deallocate local pointers/allocatables if needed
-          return
-        end if
-      end if
-
-      !!TODO: Is it time to output for this field? CAREFUL ABOUT > vs >= HERE
-      !--- The fields send out within openmp parallel region will be written out in
-      !--- diag_send_complete.
-
-      !!TODO: Is check to bounds of current field necessary?
-
-      !!TODO: Take care of submitted field data
-
-    enddo
+    fms_diag_accept_data = this%fms_diag_do_reduction(field_data, diag_field_id, oor_mask, field_weight, &
```

Comment:
I would suggest sending the field name in for error reporting purposes

Reply:
The field name can be obtained from the `diag_field_id`, so there is no need to pass it in